docs/md_merge.py

```diff
@@ -0,0 +1,357 @@
+
+    def to_text(self):
+        raise NotImplementedError()
+
+
+@dataclass
+class ImageLine(LinkLine):
+    title_text: str = ""
+
+    def to_text(self):
+        title_part = f' "{self.title_text}"' if self.title_text else ""
+        return f'{self.preceding_text}![{self.link_text}]({self.link}{title_part}){self.subsequent_text}'
+
+
+@dataclass
+class TextLinkLine(LinkLine):
+
+    def to_text(self):
+        return f"{self.preceding_text}[{self.link_text}]({self.link}){self.subsequent_text}"
+
+    @property
+    def is_anchor_link(self):
+        return self.link.startswith("#")
+
+
+def parse_line(line: str):
+    header_search = re.search(MD_HEADER_REGEX, line, re.IGNORECASE)
+    if header_search:
+        return HeaderLine(depth=len(header_search.group(2)), text=header_search.group(4))
+    image_search = re.search(MD_IMAGE_REGEX, line, re.IGNORECASE)
+    if image_search:
+        return ImageLine(
+            link=image_search.group(5),
+            link_text=image_search.group(3),
+            title_text=image_search.group(6)[1:-1],
+            preceding_text=image_search.group(1),
+            subsequent_text=image_search.group(8)
+        )
+    link_search = re.search(MD_LINK_REGEX, line, re.IGNORECASE)
+    if link_search:
+        return TextLinkLine(
+            link=link_search.group(5),
+            link_text=link_search.group(3),
+            preceding_text=link_search.group(1),
+            subsequent_text=link_search.group(7)
+        )
+    return line
+
+
+def get_dir_content(dir_path: Path, language: str = "en", files_only=False):
+    """lists the contents of a directory and sorts it alphabetically
+    starting with files and then directories
+    if files_only is set to True, directories will be excluded
+
+    tries to fetch only markdown files with a language specific suffix
+
+    Example:
+
+        document-spec.en.md
+        document-usage.en.md
+        document
+        vulnerabilities-spec.en.md
+        vulnerabilities-usage.en.md
+        vulnerabilities
+
+    the results later on influence the ordering of content in a merged file
+    """
+
+    dirs = [p for p in os.listdir(dir_path) if os.path.isdir(dir_path / p)]
+    files = get_md_language_files(dir_path, language)
+
+    if files_only:
+        return sorted(files)
+
+    result = []
+    for d in sorted(dirs):
+        for f in sorted(files):
+            if f.startswith(d):
+                result.append(f)
+        result.append(d)
+    for f in sorted(files):
+        if f not in result:
+            result.append(f)
+
+    return result
+
+
+LANGUAGE_MD_FILE_NAME_PATTERN = "(.+)[.](.{2,3})[.]md"
+
+
+def get_md_language_files(dir_path: Path, language: str = "en"):
+    """tries to fetch all markdown files with a language specific suffix
+    assumes that files in English (suffix '.en.md') are all present
+    and uses these as fallback"""
+    lang_to_file_prefix = defaultdict(list)
+    for md_file in os.listdir(dir_path):
```

Comment:
Why not use the corresponding method on the `dir_path`?

Reply:
Updated to get rid of `os` altogether.